app/helpers/ontologies_helper.rb

```diff
@@ -652,17 +652,28 @@ def ontology_edit_button
   end
 
   def upload_ontology_button
-    if session[:user].nil?
-      render Buttons::RegularButtonComponent.new(id: "upload-ontology-button", value: t('home.ontology_upload_button'), variant: "secondary", state: "regular", href: "/login?redirect=/ontologies/new") do |btn|
-        btn.icon_left do
-          inline_svg_tag "upload.svg"
-        end
-      end
-    else
-      render Buttons::RegularButtonComponent.new(id: "upload-ontology-button", value: t('home.ontology_upload_button'), variant: "secondary", state: "regular", href: new_ontology_path) do |btn|
-        btn.icon_left do
-          inline_svg_tag "upload.svg"
-        end
+    return if $READ_ONLY_PORTAL && !session[:user]&.admin?
```

Comment:
use the created helper here too

Reply:
and add this sort of checks, like this https://github.com/ontoportal-lirmm/bioportal_web_ui/blob/master/app/controllers/ontologies_controller.rb#L27